Improve await description (#22229)
mdn/content: files/en-us/web/javascript/reference/operators/await/index.md

If the promise is rejected, the `await` expression throws the rejected value. The function containing the `await` expression will [appear in the stack trace](#improving_stack_trace) of the error. Otherwise, if the rejected promise is not awaited or is immediately returned, the caller function will not appear in the stack trace.

The `expression` is resolved in the same way as {{jsxref("Promise.resolve()")}}: it's always converted to a native `Promise` and then awaited. If the `expression` is a:

- Native `Promise` (tested with `expression.constructor === Promise`): The promise is directly used and awaited natively, without calling `then()`.
- [Thenable object](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenables) (including non-native promises, polyfill, proxy, child class, etc.): A new promise is constructed with the native [`Promise()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise) constructor by calling the object's `then()` method and passing in a handler that calls the `resolve` callback.
- Non-thenable value: An already-fulfilled `Promise` is constructed and used.

> Even when the used promise has been already fulfilled, the async function execution still pauses until the next tick. In the meantime, the caller of the async function resumes execution. [See example below.](#control_flow_effects_of_await)

Reviewer: ```suggestion
Even when the used promise is already fulfilled, the async function's execution still pauses until the next tick. In the meantime, the caller of the async function resumes execution. [See example below.](#control_flow_effects_of_await)
```

// response will be "default response" if the promise is rejected
```

> Note that `catch()`'s behavior could be unexpected when the object returned by `promisedFunction()` is not a native {{jsxref("Promise")}}. To avoid unexpected behavior, use `Promise.resolve(promisedFunction())` instead of `promisedFunction()`.

Reviewer: Could you elaborate on this? You mean the returned thenable didn't implement `catch()` or implemented it in a different way? I don't think that happens a lot and here we are talking about native promises, so I suggest removing this.

Author: Okay.
Also, is "native" the correct word? Should I use "built-in" instead?

Reviewer: Both are fine for me. I slightly prefer "native".
